impl/src/main/java/org/jboss/weld/config/ConfigurationKey.java

```diff
@@ -257,6 +261,12 @@
     @Description("If a non-empty string and development mode is enabled, the Probe data will be automatically exported after deployment validation. The value represents a path of the directory where to export the data file.")
     PROBE_EXPORT_DATA_AFTER_DEPLOYMENT("org.jboss.weld.probe.exportDataAfterDeployment", ""),
 
+    /**
+     * If set to true, {@link Bootstrap#endInitialization()} must be called after all EE components which support injection are installed (that means all
+     * relevant {@link ProcessInjectionTarget} events were already fired). This allows to perform effective cleanup and further optimizations.
+     */
+    END_INIT_AFTER_EE_COMPONENTS("org.jboss.weld.bootstrap.endInitAfterEEComponents", false, true),
```

Comment:
This option basically means Weld will attempt a cleanup; shouldn't the name reflect it?
Something like `CLEANUP_AFTER_INIT` with javadoc being:
`If set to true, the integrator confirms that {@link Bootstrap#endInitialization()} is invoked after all EE components supporting injection were installed (e.g. all their ProcessInjectionTarget events were fired). This allows Weld to perform a cleanup on metadata which are no longer needed as well as other minor optimizations.`

And even if we don't go for this I would like to stress you *integrator's responsibility* for this. In a sense that this optimization should not be used unless you made sure it makes sense.

Reply:
In fact, weld already does a lot of cleanup during `endInitialization()` so this merely a hint that a more thorough cleanup can be performed.

> I would like to stress you integrator's responsibility for this...

This config property can only be set through `ExternalConfiguration`.

Reply:
Maybe something like `ALLOW_OPTIMIZED_CLEANUP`?

Reply:
Yes, sounds better.